server/CVCMIServer.cpp

```diff
@@ -1008,7 +1008,7 @@ HeroTypeID CVCMIServer::nextAllowedHero(PlayerColor player, HeroTypeID initial,
 void CVCMIServer::optionNextBonus(PlayerColor player, int dir)
 {
 	PlayerSettings & s = si->playerInfos[player];
-	PlayerStartingBonus & ret = s.bonus = static_cast<PlayerStartingBonus>(static_cast<int>(s.bonus) + dir);
+	auto ret = s.bonus = static_cast<PlayerStartingBonus>(static_cast<int>(s.bonus) + dir);
```

Comment:
```suggestion
	s.bonus = static_cast<PlayerStartingBonus>(static_cast<int>(s.bonus) + dir);
	PlayerStartingBonus & ret = s.bonus;
```

Reply:
In general there was an idea that auto & should be always used instead of auto where we expect reference.
Replacing reference with auto may cause bugs because it will no longer be reference but copy instead.